### PR TITLE
openblas: update to 0.3.25

### PR DESCRIPTION
**Pull request by szhorvat**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 13.6 22G120 arm64
Xcode 15.0.1 15A507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by catap — APPROVED** [review]

**Comment by szhorvat** [comment]
@catap This just came out, and I was hoping it'd improve the Xcode 15 situation (since it's mentioned in their changelog). But unfortunately the build fails on my macOS 13.6 system with Xcode 15. 

I get this:


```
OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 ./sblat3 < ./sblat3.dat
OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 ./dblat3 < ./dblat3.dat

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:

Program received signal SIGBUS: Access to an undefined portion of a memory object.

Backtrace for this error:
```

Then the build hangs on the `dblat3` process.

**Comment by catap** [comment]
@szhorvat but github CI had suceffuly build it!

**Comment by szhorvat** [comment]
Glad to see that! Maybe the problem is that I'm on arm64?

The hang and crashes occur during testing. @catap, did you set it up to run tests even when just building the library, or did the OpenBLAS folks do this?

Unfortunately I don't think I have the time (or expertise) to debug this properly, so at this point I'd like to hand over the PR. Let me know if you want me to close it, or to mark it as ready for review.

**Comment by catap** [comment]
@szhorvat I'll take a look on this later today. May I ask you to grand me permission to push to your repo to avoid creating a new PR / asking you to assume anything into this one.

**Comment by szhorvat** [comment]
@catap I added you temporarily until this is fixed.

**Comment by catap** [comment]
@szhorvat I suggest to mark this PR as ready to review. I can't reproduce your issue on my machine :(

**Comment by szhorvat** [comment]
Did you see this issue? https://trac.macports.org/ticket/68689 Any tips? Can you reproduce it?

**Comment by catap** [comment]
This one I can reproduce!

The fix:
```diff
diff --git a/Makefile.install b/Makefile.install
index 81f959177..29b623ff5 100644
--- a/Makefile.install
+++ b/Makefile.install
@@ -42,12 +42,12 @@ lib.grd :
        $(error OpenBLAS: Please run "make" firstly)
 
 install :      lib.grd
-       @-mkdir -p "$(DESTDIR)$(PREFIX)"
-       @-mkdir -p "$(DESTDIR)$(OPENBLAS_INCLUDE_DIR)"
-       @-mkdir -p "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
-       @-mkdir -p "$(DESTDIR)$(OPENBLAS_BINARY_DIR)"
-       @-mkdir -p "$(DESTDIR)$(OPENBLAS_CMAKE_DIR)"
-       @-mkdir -p "$(DESTDIR)$(OPENBLAS_PKGCONFIG_DIR)"
+       mkdir -p "$(DESTDIR)$(PREFIX)"
+       mkdir -p "$(DESTDIR)$(OPENBLAS_INCLUDE_DIR)"
+       mkdir -p "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
+       mkdir -p "$(DESTDIR)$(OPENBLAS_BINARY_DIR)"
+       mkdir -p "$(DESTDIR)$(OPENBLAS_CMAKE_DIR)"
+       mkdir -p "$(DESTDIR)$(OPENBLAS_PKGCONFIG_DIR)"
        @echo Generating openblas_config.h in $(DESTDIR)$(OPENBLAS_INCLUDE_DIR)
 #for inc
        @echo \#ifndef OPENBLAS_CONFIG_H > "$(DESTDIR)$(OPENBLAS_INCLUDE_DIR)/openblas_config.h"
```

**Comment by szhorvat** [comment]
Can you include it in this PR please? You have commit access to my repo.

**Comment by catap** [comment]
I've pushed mine patch. I not sure that it should be backported to upstream, I really think that this port should be rework to cmake which should fix that issue on the first place.

Thus, your issue seems to be quite similar to original failure on Xcode 15 and I suggest to report that in related issue at upstream: https://github.com/OpenMathLib/OpenBLAS/issues/4239

**Comment by pmetzger** [comment]
Should I merge this?

**Comment by szhorvat** [comment]
Up to @catap from this point. As I said, installation won't complete on my M2 Mac, and I wasn't able to fix it. 

OpenBLAS is essential for lots of ports so:

 - we really shouldn't break it
 - CMake-based ports that link to OpenBLAS are broken at the moment, and the fix is in this PR ... so that would be urgent to merge (see https://trac.macports.org/ticket/68689)

**Pull request merged**
